inc/http-client/class-cache-strategy.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace RemoteDataBlocks\HttpClient;
+
+use Kevinrob\GuzzleCache\Strategy\GreedyCacheStrategy;
+use Psr\Http\Message\RequestInterface;
+use RemoteDataBlocks\Logging\LoggerManager;
+
+class CacheStrategy extends GreedyCacheStrategy {
+	const HEADER_TTL          = 'X-RemoteDataBlocks-TTL';
+	const HEADER_BYPASS_CACHE = 'X-RemoteDataBlocks-Bypass-Cache';
+
+	private static function getRequestString( RequestInterface $request ) {
+		$uri        = $request->getUri();
+		$uri_string = $uri->getScheme() . '://' . $uri->getHost() . $uri->getPath();
+		$method     = $request->getMethod();
+		$body       = $request->getBody()->getContents();
+		return $method . ' ' . $uri_string . ' ' . $body;
+	}
+
+	protected function should_bypass_cache( RequestInterface $request ) {
+		$bypass = $request->getHeader( self::HEADER_BYPASS_CACHE );
+		if ( ! empty( $bypass ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	public function fetch( RequestInterface $request ) {
+		$logger = LoggerManager::instance();
+
+		if ( $this->should_bypass_cache( $request ) ) {
+			$logger->debug( 'Bypassing cache: ' . self::getRequestString( $request ) );
```

Comment:
Should we have any prefix in debugging messages which will make the search easier?

Reply:
Good idea.

Reply:
Just pass a string to `LoggerManager::instance()`, it accepts a namespace

Reply:
@chriszarate Seems like the `Logger` constructor does, but not the static instance fn:

https://github.com/Automattic/remote-data-blocks/blob/01eaa598eb983ec9aa9e98a586aa0aa4864750f6/inc/logging/logger-manager/logger-manager.php#L27-L33

...unless I'm missing something?

Reply:
Oh, right. I guess we already have a namespace and don't really need one? Or we want multi-level namespaces?

Reply:
I think @mehmoodak's suggestion was for multi-level namespaces to be able to more-easily identify that the debug log came from e.g. the HTTP Client module (similar to how the [debug npm package](https://www.npmjs.com/package/debug) allows multi-level prefixing / filtering.

I think I'll file it to an issue for an enhancement outside this PR.

Reply:
https://github.com/Automattic/remote-data-blocks/issues/12